include/OpenSpaceToolkit/Physics/Environment/Atmospheric/Earth/Weather/CSSISpaceWeather.hpp

```diff
@@ -0,0 +1,172 @@
+#include <OpenSpaceToolkit/Physics/Time/Interval.hpp>
+
+namespace ostk
+{
+namespace physics
+{
+namespace environment
+{
+namespace atmospheric
+{
+namespace earth
+{
+namespace weather
+{
+
+namespace fs = ostk::core::fs;
+
+using ostk::core::types::Integer;
+using ostk::core::types::Real;
+using ostk::core::types::String;
+using ostk::core::ctnr::Map;
+using ostk::core::ctnr::Array;
+using ostk::core::fs::File;
+
+using ostk::physics::time::Date;
+using ostk::physics::time::DateTime;
+using ostk::physics::time::Instant;
+using ostk::physics::time::Duration;
+using ostk::physics::time::Interval;
+using ostk::physics::time::Scale;
+
+/// @brief                      Center for Space Weather and Innovation (CSSI) Space Weather data file.
+///
+///                             Consolidated data set which contains solar radiation and geomagnetic indices.
+///                             Particularly contains the F10.7 solar flux index and Ap/Kp geomagnetic indices,
+///                             which are commonly used to model atmospheric density.
+///
+/// @ref                        http://celestrak.org/SpaceData/SpaceWx-format.php
+
+class CSSISpaceWeather
+{
+   public:
+    struct Observation
+    {
+        Date date;     /// UTC day of reading.
+        Integer BSRN;  /// Bartels Solar Rotation Number. A sequence of 27-day intervals counted continuously from 1832
+                       /// Feb 8.
+        Integer ND;    /// Number of Day within the Bartels 27-day cycle (01-27).
+        Integer KP1;   /// Planetary 3-hour Range Index (Kp) for 0000-0300 UT.
+        Integer KP2;   /// Planetary 3-hour Range Index (Kp) for 0300-0600 UT.
+        Integer KP3;   /// Planetary 3-hour Range Index (Kp) for 0600-0900 UT.
+        Integer KP4;   /// Planetary 3-hour Range Index (Kp) for 0900-1200 UT.
+        Integer KP5;   /// Planetary 3-hour Range Index (Kp) for 1200-1500 UT.
+        Integer KP6;   /// Planetary 3-hour Range Index (Kp) for 1500-1800 UT.
+        Integer KP7;   /// Planetary 3-hour Range Index (Kp) for 1800-2100 UT.
+        Integer KP8;   /// Planetary 3-hour Range Index (Kp) for 2100-0000 UT.
+        Integer KP_SUM;  /// Sum of the 8 Kp indices for the day.
+        Integer AP1;     /// Planetary Equivalent Amplitude (Ap) for 0000-0300 UT.
+        Integer AP2;     /// Planetary Equivalent Amplitude (Ap) for 0300-0600 UT.
+        Integer AP3;     /// Planetary Equivalent Amplitude (Ap) for 0600-0900 UT.
+        Integer AP4;     /// Planetary Equivalent Amplitude (Ap) for 0900-1200 UT.
+        Integer AP5;     /// Planetary Equivalent Amplitude (Ap) for 1200-1500 UT.
+        Integer AP6;     /// Planetary Equivalent Amplitude (Ap) for 1500-1800 UT.
+        Integer AP7;     /// Planetary Equivalent Amplitude (Ap) for 1800-2100 UT.
+        Integer AP8;     /// Planetary Equivalent Amplitude (Ap) for 2100-0000 UT.
+        Integer AP_AVG;  /// Arithmetic average of the 8 Ap indices for the day.
```

Comment:
Yeah the idea was to match the CSV, which uses all caps snake case. (@Derollez that link is for the legacy format, this uses the CSV format described [here](https://celestrak.org/SpaceData/SpaceWx-format.php)). 

Honestly, I meant to revisit this but it slipped. I think CamelCase as Remy pointed out looks a lot cleaner (especially to indicate the subscripts). Will update!

Reply:
Alright clang format doesn't allow everything to be fully lined up, but since renaming the variable, the comments are _mostly_ lined up. (although add a `///<` makes the alignment very wonky. Current version looks pretty clean. Think it's good enough?